update(rules catalog): introduce sigma (#157)
SEKOIA-IO/documentation: docs/operation_center/rules_catalog.md

  
  For CTI rules, you just have to select the source of the indicators: SEKOIA Intelligence Feed is an IOCs feed managed by SEKOIA's Purple Team (indicators present in the SEKOIA.IO Intelligence Center).
  

>   For correlation rules, SEKOIA.IO supports two languages : SIGMA and STIX Patterning. More details about this langage are given [below](rules_catalog.md#sigma).

Reviewer: ```suggestion
  For correlation rules, SEKOIA.IO supports two languages : [SIGMA](rules_catalog.md#sigma) and [STIX Patterning](rules_catalog.md#stix-patterning). More details about these languages are given below.```

!!! note
    Modification of rules parameters will be applied for new alerts, raised after the compilation of the rule.

## Sigma Format
Sigma is a generic and open format available to write correlation rules. 
The rule format is easy to write and applicable to any field available in ECS in the platform. The langage used is YAML.

The structure is the following :
```
detection:
  <Search-Idenfier>
    <string-list>
    <field:value>

>   <Condition>

Reviewer: ```suggestion
  condition: <Condition>
```

``` 

`<Search-Identifier>` is a unique identifier that will be used in `<Condition>`. A `<Search-Idenfier>` can contain two different structures, lists and maps.

!!! Note
    Currently, we set a limitation and rules can be written for only one event. Aggregation expressions are not supported yet.
    
For the full specification of SIGMA, please refers to the [SIGMA Github](https://github.com/SigmaHQ/sigma/wiki/Specification)

### Lists
A list contains one or more strings that will match the full event log. Multiple strings will be linked with a logicial `OR`.

**Example** : Raise an alert when an event contains "DangerousThreat.exe" or "powershell.exe malicousfile.ps1"
```
detection:
  keywords:
    - DangerousThreat.exe
    - powershell.exe maliciousfile.ps1

> condition: keywords

Reviewer: ```suggestion
  condition: keywords
```
